master/buildbot/process/build.py

```diff
@@ -225,23 +225,32 @@ def setupPropertiesKnownBeforeBuildStarts(props, requests, builder,
         # get worker properties
         # navigate our way back to the L{buildbot.worker.Worker}
         # object that came from the config, and get its properties
-        workerforbuilder.worker.setupProperties(props)
+        if workerforbuilder is not None:
+            workerforbuilder.worker.setupProperties(props)
 
-    def setupOwnProperties(self):
+    @staticmethod
+    def setupBuildProperties(props, requests, sources=None):
         # now set some properties of our own, corresponding to the
         # build itself
-        props = self.getProperties()
-        props.setProperty("buildnumber", self.number, "Build")
+        props.setProperty("buildnumber", None, "Build")
+
+        if sources is None:
+            sources = requests[0].mergeSourceStampsWith(requests[1:])
 
-        if self.sources and len(self.sources) == 1:
+        if sources and len(sources) == 1:
             # old interface for backwards compatibility
-            source = self.sources[0]
+            source = sources[0]
             props.setProperty("branch", source.branch, "Build")
             props.setProperty("revision", source.revision, "Build")
             props.setProperty("repository", source.repository, "Build")
             props.setProperty("codebase", source.codebase, "Build")
             props.setProperty("project", source.project, "Build")
 
+    def setupOwnProperties(self):
+        props = self.getProperties()
```

Comment:
I guess we no longer need this setupOwnProperties() function as it's used in a single place.

Reply:
Just to make sure I got this one. You want to remove the method completely (thus breaking the API) right?

Reply:
Yes. Buildbot is not exposing its internals as a real API, it's considered an implementation detail. I don't see this function documented and Buildbot has a policy of non-documented == internal detail, thus we can do whatever we want with it.

Reply:
There are a couple of tests that you'll need changing, but it's just a simple function inlining. Though now that the functions themselves are static it may make sense to greatly simplify the test by avoiding a reference to a real build altogether (but you don't need to do that in this PR).